Backend/app/therapy_sessions/serializers.py

```
@@ -0,0 +1,738 @@
+        ]
+    
+    def validate_patient_id(self, value):
+        """Validate that patient exists and is connected to therapist"""
+        try:
+            patient = User.objects.get(id=value, user_type='patient')
+            therapist = self.context['request'].user
+            if not patient.patient_profile.therapist or patient.patient_profile.therapist.user != therapist:
+                raise serializers.ValidationError("Patient is not connected to this therapist.")
+            return value
+        except User.DoesNotExist:
+            raise serializers.ValidationError("Patient not found.")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)